src/Reflection/Adapter/ReflectionIntersectionType.php

```diff
@@ -6,25 +6,26 @@
 
 use ReflectionIntersectionType as CoreReflectionIntersectionType;
 use Roave\BetterReflection\Reflection\ReflectionIntersectionType as BetterReflectionIntersectionType;
-use Roave\BetterReflection\Reflection\ReflectionNamedType as BetterReflectionNamedType;
-use Roave\BetterReflection\Reflection\ReflectionUnionType as BetterReflectionUnionType;
+use Roave\BetterReflection\Reflection\ReflectionType as BetterReflectionType;
 
-use function array_filter;
 use function array_map;
+use function assert;
 
 class ReflectionIntersectionType extends CoreReflectionIntersectionType
 {
     public function __construct(private BetterReflectionIntersectionType $betterReflectionType)
     {
     }
 
-    /** @return array<ReflectionNamedType> */
+    /** @return list<ReflectionNamedType> */
     public function getTypes(): array
     {
-        return array_filter(
-            array_map(static fn (BetterReflectionNamedType|BetterReflectionUnionType|BetterReflectionIntersectionType $type): ReflectionNamedType|ReflectionUnionType|ReflectionIntersectionType|null => ReflectionType::fromTypeOrNull($type), $this->betterReflectionType->getTypes()),
-            static fn (ReflectionNamedType|ReflectionUnionType|ReflectionIntersectionType|null $type): bool => $type instanceof ReflectionNamedType,
-        );
+        return array_map(static function (BetterReflectionType $type): ReflectionNamedType {
```

Comment:
`BetterReflectionType` is not `final`: IMO, the union type should be preserved :thinking:

Reply:
Changed to `BetterReflectionNamedType` because `ReflectionIntersectionType::getTypes()` returns `list<ReflectionNamedType>`

Reply:
I see :thinking: 

Need to reason about this a bit, 'bit slow after processing a few hundred mails, sorry.